### PR TITLE
PPAL-23:  Set up User redux, graphql, and a lot of foundation

### PR DESCRIPTION
**Pull request by arfusop-dev**

### CODE REVIEW

File: src/lib/gql/mutations/user.ts [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+      ... on User {
+        _id
+        email
+        firstName
+        lastName
+        providerId
+        avatar
+        createdAt
+        lastLogin
+        newUser
```
Comment (KevinLaCarrubba) [inline src/lib/gql/mutations/user.ts:48]: Do you need tier here ?

File: src/lib/gql/queries/user.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+      ... on User {
+        _id
+        email
+        firstName
+        lastName
+        providerId
+        avatar
+        createdAt
+        lastLogin
+        newUser
```
Comment (KevinLaCarrubba) [inline src/lib/gql/queries/user.ts:15]: And tier here ?

File: src/lib/redux/slices/user.ts [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+export const initialState = {
+  profile: {
+    _id: '',
+    email: '',
+    firstName: '',
+    lastName: '',
+    avatar: '',
+    providerId: '',
+    lastLogin: null,
+    newUser: false
```
Comment (KevinLaCarrubba) [inline src/lib/redux/slices/user.ts]: Idk about the tier anymore

File: src/lib/gql/mutations/user.ts [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+        tier
```
Comment (KevinLaCarrubba) [inline src/lib/gql/mutations/user.ts:26]: Is this the wrong tier then ?

**Review by KevinLaCarrubba — APPROVED** [review]
Idk what tier is rigght or
Missing 🤣😂

File: src/lib/gql/mutations/user.ts (hunk 4 above)
Comment (arfusop-dev) [reply src/lib/gql/mutations/user.ts:26]: Nope you right. This is the only spot thats correct

File: src/lib/gql/mutations/user.ts (hunk 1 above)
Comment (arfusop-dev) [reply src/lib/gql/mutations/user.ts:48]: Added

**Pull request merged**